plugins/by-name/treesitter/default.nix

```diff
@@ -189,135 +147,45 @@ lib.nixvim.plugins.mkNeovimPlugin {
-        a lua function that returns a boolean indicating the parser should be disabled.
-      '';
-
-      custom_captures = lib.nixvim.defaultNullOpts.mkAttrsOf types.str { } ''
-        Custom capture group highlighting.
-      '';
-    };
-
-    incremental_selection = {
-      enable = lib.nixvim.defaultNullOpts.mkBool false ''
-        Incremental selection based on the named nodes from the grammar.
-      '';
-
-      keymaps =
-        let
-          mkKeymap =
-            default:
-            lib.nixvim.defaultNullOpts.mkNullableWithRaw (
-              with types; either str bool
-            ) default "Key shortcut or false to unset.";
-        in
-        {
-          init_selection = mkKeymap "gnn";
-          node_incremental = mkKeymap "grn";
-          scope_incremental = mkKeymap "grc";
-          node_decremental = mkKeymap "grm";
-        };
-    };
-
-    indent = {
-      enable = lib.nixvim.defaultNullOpts.mkBool false ''
-        Whether to enable treesitter indentation.
-      '';
-    };
-
-    ensure_installed = lib.nixvim.defaultNullOpts.mkNullable' {
-      type =
-        with lib.types;
-        oneOf [
-          (enum [ "all" ])
-          (listOf (maybeRaw str))
-          rawLua
-        ];
-      pluginDefault = [ ];
-      description = ''
-        Either `"all"` or a list of languages to ensure installing.
-      '';
-    };
-
-    ignore_install = lib.nixvim.defaultNullOpts.mkListOf types.str [ ] ''
-      List of parsers to ignore installing. Used when `ensure_installed` is set to `"all"`.
-    '';
-
-    parser_install_dir = lib.nixvim.mkNullOrOption' {
-      type = with lib.types; maybeRaw str;
-      # Backport the default from nvim-treesitter 1.0
-      # The current default doesn't work on nix, as it is readonly
-      default.__raw = "vim.fs.joinpath(vim.fn.stdpath('data'), 'site')";
-      pluginDefault = lib.literalMD "the plugin's package directory";
-      description = ''
-        Location of the parsers to be installed by the plugin (only needed when `nixGrammars` is disabled).
-
-        By default, parsers are installed to the "site" dir.
-        If set to `null` the _plugin default_ is used, which will not work on nix.
-      '';
-    };
-
-    sync_install = lib.nixvim.defaultNullOpts.mkBool false ''
-      Install parsers synchronously (only applied to `ensure_installed`).
-    '';
+  settingsExample = {
+    install_dir = lib.nixvim.nestedLiteralLua "vim.fs.joinpath(vim.fn.stdpath('data'), 'site')";
   };
 
-  settingsExample = {
-    auto_install = false;
-    ensure_installed = "all";
-    ignore_install = [ "rust" ];
-    parser_install_dir.__raw = "vim.fs.joinpath(vim.fn.stdpath('data'), 'treesitter')";
-    sync_install = false;
+  extraOptions = {
+    folding =
+      let
+        foldingSubmodule = types.submodule {
+          options = {
+            enable = lib.mkEnableOption "tree-sitter based folding";
+          };
+        };
+      in
```

Comment:
Nit: moving this let-block to the `type =` binding will allow the rest of the option declaration to be unindented (i.e. `folding = mkOption {`).